### PR TITLE
feat: skipTake

### PR DESCRIPTION
**Pull request by damiangreen**
Add `skipTake` function. Solves https://github.com/char0n/ramda-adjunct/issues/1237

### CODE REVIEW

**Comment by char0n** [comment]
By a quick peak, I can tell we're missing tests for typings.

**Comment by damiangreen** [comment]
yep, i had a crack at those, but  the types didnt work, i think because of the underlying type system in ramda

File: types/test/skipTake.ts [hunk 1]
```diff
@@ -0,0 +1,4 @@
+import * as RA from 'ramda-adjunct';
+
+RA.skipTake(2, [1, 2, 3, 4]); // $ExpectType number[]
+RA.skipTake<number>(2)([1, 2, 3, 4]); // $ExpectType number[]
```
Comment (char0n) [inline types/test/skipTake.ts]: Let's add some error expectations. Example here: https://github.com/char0n/ramda-adjunct/blob/4d0e49a067439be21ddb525cc8a0d5090395421a/types/test/fnull.ts#L9-L8

**Comment by damiangreen** [comment]
all changes made, let us know what you think

File: types/index.d.ts [hunk 2]
```diff
@@ -2,1360 +2,1472 @@ declare var RA: RamdaAdjunct.Static;
 // TypeScript Version: 2.4
 
 declare namespace RamdaAdjunct {
```
Comment (char0n) [inline types/index.d.ts]: This whole file have been changed. We only want changes regarding `skipTake`.

File: test/skipTake.js [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+      assert.deepEqual(RA.skipTake(0, []), []);
+    });
+  });
+
+  it('should be curried', function() {
+    assert.deepEqual(RA.skipTake(2)([1, 2, 3, 4]), [1, 3]);
+    assert.deepEqual(RA.skipTake(2, [1, 2, 3, 4]), [1, 3]);
+  });
+
+  it('should support placeholder to specify "gaps"', function() {
```
Comment (char0n) [inline test/skipTake.js]: We can drop this last test. `should be curried` is already testing support for placeholder implicitly.

File: test/skipTake.js [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+      assert.deepEqual(RA.skipTake(3, []), []);
+    });
+  });
+
+  context('given negative count as input', function() {
+    specify('should return an empty array', function() {
+      assert.deepEqual(RA.skipTake(-1, []), []);
+    });
+  });
+
```
Comment (char0n) [inline test/skipTake.js]: I'm still missing test for negative skip and non empty array. Not sure how the function behave in that scenario according to the tests.

Comment (damiangreen) [reply test/skipTake.js]: negative skip? I'm not sure i follow that one

Comment (damiangreen) [reply test/skipTake.js]: the first test test for a non empty array?

File: test/skipTake.js (hunk 3 above)
Comment (damiangreen) [reply test/skipTake.js]: pushed.

File: types/index.d.ts (hunk 2 above)
Comment (damiangreen) [reply types/index.d.ts]: fixed

File: test/skipTake.js (hunk 4 above)
Comment (char0n) [reply test/skipTake.js]: I'm talking about this usecase. I'm not sure what will be the result of this.
```js
RA.skipTake(-2, [1, 2, 3, 4])
```

Comment (damiangreen) [reply test/skipTake.js]: very good point.. and right now it returns [1,3].
What do you believe should be the correct behaviour here? throw an exception or return an empty array?

Comment (char0n) [reply test/skipTake.js]: I believe what we see is expected behavior. I'd just create a tests which shows that it doesn't really matter if the number is positive or negative integer.

Another thing that comes to my mind is how does the function reacts on providing `2.3` number instead of `2`?

Comment (char0n) [reply test/skipTake.js]: Tests tests tests...they gives all the answers ;]
